ecosystem/indexer-grpc/indexer-grpc-parser/README.md

```diff
@@ -0,0 +1,126 @@
+
+### Start the indexer fullnode
+1. Modify fullnode.yaml with your details. Example fullnode.yaml modification:
+   ```
+   storage:
+   enable_indexer: true
+   # This is to avoid the node being pruned
+   storage_pruner_config:
+      ledger_pruner_config:
+         enable: false
+
+   indexer:
+      enabled: true
+      postgres_uri: "postgres://postgres@localhost:5432/postgres"
+      processor: "default_processor"
+      check_chain_id: true
+      emit_every: 500
+   ``` 
+2. Run the indexer fullnode:
+    ```bash
+    cargo run -p aptos-node --features "indexer" --release -- -f <some_path>/fullnode.yaml
+    ```
+
+### Optional pgadmin4
+1. Complete the [steps](#installation-guide-for-apple-sillicon) above.
+2. Install pgadmin4: `brew install --cask pgadmin4`
+3. Open PgAdmin4.
+4. Create a master password.
+5. Right Click Servers > `Register` > `Server`
+6. Enter the following information in the registration Modal:  
+    ```yaml
+    General:
+    Name: Indexer
+
+    Connection:
+    Hostname / Address: 127.0.0.1
+    Port: 5432
+    Maintenance Database: postgres
+    Username: postgres
+    ```
+7. Save.
+
+> *Notes*:
+> - Diesel uses the `DATABASE_URL` env var to connect to the database, or the `--database-url` argument.
+> - Diesel CLI can be installed via cargo. For example: `cargo install diesel_cli --no-default-features --features postgres`
+> - `diesel migration run` sets up the database and runs all available migrations.
+> - Aptos tests use the `INDEXER_DATABASE_URL` env var. It needs to be set for the relevant tests to run.
+> - Postgres can be [installed and run via brew](https://wiki.postgresql.org/wiki/Homebrew).
+
+## Adding new tables / updating tables with Diesel
+
+* `diesel migration generate <your_migration_name>` generates a new folder containing `up.sql + down.sql` for your
+  migration
+* `diesel migration run` to apply the missing migrations. This will re-generate `schema.rs` as required.
+* `diesel migration redo` to rollback and apply the last migration
+* `diesel database reset` drops the existing database and reruns all the migrations
+* You can find more information in the [Diesel](https://diesel.rs/) documentation.
+
+### Miscellaneous
+1. If you run into
+   ```bash
+     = note: ld: library not found for -lpq
+             clang: error: linker command failed with exit code 1 (use -v to see invocation)
+   ```
+first make sure you have `postgres` and `libpq` installed via `homebrew`, see installation guide above for more details.
+This error is regarding the `libpq` library, a postgres C API library which diesel needs to run, more on this issue [here](https://github.com/diesel-rs/diesel/issues/2612).  
+
+2. [PostgreSQL Mac M1 installation guide](https://gist.github.com/phortuin/2fe698b6c741fd84357cec84219c6667).
```

Comment:
@bowenyang007 Does this also apply to M2 Macs?

Reply:
lol no idea. by the way this readme will be completely re-written. You might want to change the existing indexer readme instead.
https://github.com/aptos-labs/aptos-core/blob/aptos-grpc-indexer-client/crates/indexer/README.md